Updated appearance of modal

OlliverLar opened the pull request:
Har fixat så att innehållet i steg 2 är "fallande". Samt fixat så att modalen har en bakgrund med lägre opacitet liknande menyn för mobil

OlliverLar committed:
Updated appearance of modal

saeraa commented on 02-esc/modal.js:
```diff
@@ -101,22 +103,27 @@ function possibleNumberOfParticipants(min, max) {
             <select class="input-field" name="participants" id="participants">
             </select>
-            <input class="button primary open-modal-step-3" type="submit" value="Submit booking">
+            <input id="secondButton" class="button primary open-modal-step-3" type="submit" value="Submit booking">
```
Knappen ska vara nere i högra hörnet enligt spec.

OlliverLar replied:
Fixat!

saeraa reviewed (changes requested):
Bra början, men jag tycker inte den är så väldigt  lik specen i Whimsical?
![2022-11-29](https://user-images.githubusercontent.com/99484730/204585431-1fb11b2f-bead-4181-8f54-17dada60ec3e.png)

Till exempel har du label och sedan input på separata rader. Knappen är nere i hörnet. Det är inget : efter Namn

Dessutom ska det väl faktiskt stå titeln på rummet istället för "Title of room"? 😉

OlliverLar committed:
Updated modal

- Ändrat utseendet på modalen för att likna specen på Whimsical.

- Fixat så att titeln på rummet är korrekt.

- Fixat så att det inte går att scrolla på "bakgrunden" medans modalen är öppen.

OlliverLar commented:
Har gjort en ny commit där jag tror att jag åtgärdat allt det som nämndes.

saeraa commented:
Snyggt jobbat! Enda kommentaren jag har nu är att det blir väldigt "squishat" på mobilvyn, kanske kan behöva lite padding och mindre textstorlek? Eller är jag för petig nu? 😄

![2022-11-30](https://user-images.githubusercontent.com/99484730/204818133-e81b41b6-a5d8-4980-bc76-06c91d169cb6.png)

OlliverLar commented:
> Snyggt jobbat! Enda kommentaren jag har nu är att det blir väldigt "squishat" på mobilvyn, kanske kan behöva lite padding och mindre textstorlek? Eller är jag för petig nu? 😄
> 
> ![2022-11-30](https://user-images.githubusercontent.com/99484730/204818133-e81b41b6-a5d8-4980-bc76-06c91d169cb6.png)

Du har helt rätt! Fixar det!

OlliverLar committed:
Fixed padding for mobile-version

Lade till padding (prövade också att göra texten mindre men det såg i min åsikt konstigare ur)

saeraa reviewed (approved):
Snyggt! Bra jobbat 👏

richardolsson commented on 02-esc/challenges.js:
```diff
@@ -197,6 +197,7 @@ function showAllChallenges(resultFromAPI) {
 		btnBook.setAttribute("data-id", id);
 		btnBook.setAttribute("data-minparticipants", minParticipants);
 		btnBook.setAttribute("data-maxparticipants", maxParticipants);
+		btnBook.setAttribute("data-title", title);
```
Storing data like this in the DOM is unconventional, and I'm not sure what to think of it. On one hand, it blurs the lines between data and logic on one hand, and presentation on the other hand. At the same time, using `data-` attributes clearly marks the DOM attributes as containing data that is used by code, as opposed to just grabbing the data from an ID attribute or something like that.

Regardless of how to think about data in DOM _on principle_, storing data about the challenge on a button feels weird, because the data is not related to the button itself, but to the challenge as a whole.

richardolsson commented on 02-esc/modal.js:
```diff
@@ -15,27 +16,38 @@ export function openModal(e) {
 	modalDiv = document.createElement("div");
 	modalDiv.className = "modal-div";
 	modalDiv.innerHTML = `
+		<div class="modalContents">
```
There is some weird indentation going on in this code. Reading HTML strings in javascript is always difficult, but it's made more difficult by the inconsistent indentation.

richardolsson commented on 02-esc/modal.js:
```diff
@@ -15,27 +16,38 @@ export function openModal(e) {
+
+
+	if (document.querySelector(".modal-div").style.position = "fixed") {
+		document.querySelector("body").style.overflow = "hidden"
```
Nice addition!

richardolsson commented on 02-esc/modal.js:
```diff
@@ -101,22 +113,31 @@ function possibleNumberOfParticipants(min, max) {
         <form action="">
-            <label for="name">Name</label>
+            <label class="modalInputLabels" for="name">Name</label>
+			<br>
```
Line break elements (`<br>`) are rarely used nowadays, because most of the time that used to appear in olden times, they were used to achieve layout, which should not be done in HTML but with styling.

A better way of doing this would be to wrap each label/input pair in a div, or (as is common) wrap the input in the label, and make the label a `display: block` to put each on a new line.

richardolsson reviewed (commented):
There was not a ton of Javascript in this PR, but I've commented on some stuff below anyway. 👇
